lib/teiserver_web/live/account/relationship/index.ex

```diff
@@ -255,6 +265,58 @@ defmodule TeiserverWeb.Account.RelationshipLive.Index do
     {:noreply, socket}
   end
 
+  def handle_event("purge-avoids", _params, socket) do
+    userid = socket.assigns.current_user.id
+    days = get_purge_days_cutoff(socket)
+    num_rows = RelationshipLib.delete_inactive_ignores_avoids_blocks(userid, days)
+
+    socket =
+      socket |> assign(:purge_avoids_message, "#{num_rows} inactive users purged.")
+
+    {:noreply, socket}
+  end
+
+  def handle_event("purge-friends", _params, socket) do
+    days_cutoff = get_purge_days_cutoff(socket)
+
+    # Get all friends of this user
+    friends = socket.assigns[:friends]
+
+    num_friends_deleted =
+      Enum.map(friends, fn friend ->
+        last_login = friend.other_user.last_login
+        days = abs(DateTime.diff(last_login, Timex.now(), :day))
+        should_delete? = days > days_cutoff
+
+        if(should_delete?) do
+          # Delete friend
+          Account.delete_friend(friend)
+        end
+
+        should_delete?
+      end)
+      |> Enum.filter(fn deleted? -> deleted? == true end)
+      |> length()
+
+    socket =
+      socket |> assign(:purge_friends_message, "#{num_friends_deleted} inactive friends purged.")
```

Comment:
Tested with
```
      |> put_flash(:info, "Test")
```
Nothing showed up

Reply:
ah damn, this is because this is a liveview, and the flash feature is only in place for regular views :(
it would be nice to enable a flash feature for live views, but that's out of scope for this change.